src/python-frontend/python_converter.cpp

```diff
@@ -251,6 +251,26 @@ exprt python_converter::get_expr(const nlohmann::json &element)
     if(element.contains("func") && element["_type"] == "Call")
     {
       std::string func_name = element["func"]["id"];
+
+      // Non-determinism
+      auto starts_with = [](const std::string &str, const std::string &prefix) {
+        return str.compare(0, prefix.length(), prefix) == 0;
+      };
+
+      std::string nondet("nondet");
+      if(starts_with(func_name, nondet))
+      {
+        // Function name pattern: nondet_(type). e.g: nondet_bool(), nondet_int()
+        size_t underscore_pos = func_name.find("_");
+        if(underscore_pos != std::string::npos)
```

Comment:
These conditions (starts-with "nondet" and constains "_") are too lax: they would also match e.g. `nondeterministic_procedure()` but should only apply to those functions that follow the naming conventions.